### DISCUSSION TITLE
English Localization and Text Representation

### DISCUSSION THREAD

**Discussion by anatawa12** [vrc-get/vrc-get#894] [Localization]
This is discussion for English Localization and text representation.

Please feel free to suggest better localization / representation here.

The maintainer of English localization is @anatawa12 and @Sayamame-beans.

### Summary of localization updates

| locale | missing count | exceeding count | link |
| -- | -- | -- | -- |
| ja | 0 | 0 | [link](https://github.com/vrc-get/vrc-get/discussions/855) |
| de | 6 | 0 | [link](https://github.com/vrc-get/vrc-get/discussions/860) |
| zh_hans | 0 | 0 | [link](https://github.com/vrc-get/vrc-get/discussions/888) |
| fr | 6 | 0 | [link](https://github.com/vrc-get/vrc-get/discussions/909) |
| zh_hant | 0 | 0 | [link](https://github.com/vrc-get/vrc-get/discussions/1443) |
| ko | 6 | 0 | [link](https://github.com/vrc-get/vrc-get/discussions/1823) |
| es | 6 | 0 | [link](https://github.com/vrc-get/vrc-get/discussions/3122) |
| ru | 6 | 0 | [link](https://github.com/vrc-get/vrc-get/discussions/3123) |

**Comment by lu-na-r** [comment]
I found that key: `projects:dialog:warn remove project` in Japanese has its unique message: "If you remove the directory, you cannot undo this action."（ファイルを削除した場合は復元出来ません）although other supported language only says "Are you sure?". 

I wonder if this message should be standard. I you are for, I will add this in the value.

**Reply by lu-na-r** [reply]
@anatawa12 @Sayamame-beans

**Reply by anatawa12** [reply]
I think "are you sure" is enough but sayamame added that in jp localization.

In addition, in current version of ALCOM moves the folder to trash folder instead of removing permanently so It's incorrect statement now.

**Reply by Sayamame-beans** [reply]
> moves the folder to trash folder instead of removing permanently

Certainly! I forgot about that change. Hmm...

**Reply by lu-na-r** [reply]
Got it, thank you. We may as well discuss it in Japanese localisation channel instead.

**Comment by lu-na-r** [comment]
@anatawa12  @Sayamame-beans 

These three keys: `projects:manage:dialog:uninstall package as requested`, `projects:manage:dialog:uninstall package as legacy` and `projects:manage:dialog:uninstall package as unused` have a little bit different values between English and Japanese.

English:
```
    "projects:manage:dialog:uninstall package as requested": "Remove <b>{{name}}</b> as you requested",
    "projects:manage:dialog:uninstall package as legacy": "Remove <b>{{name}}</b> since it's a legacy package",
    "projects:manage:dialog:uninstall package as unused": "Remove <b>{{name}}</b> since it's not used",
```

which means
```
    "projects:manage:dialog:uninstall package as requested": "リクエストの通り、<b>{{name}}</b>を削除します。",
    "projects:manage:dialog:uninstall package as legacy": "レガシーパッケージなので、{{name}}</b>を削除します",
    "projects:manage:dialog:uninstall package as unused": "使われていないので、<b>{{name}}</b>を削除します。",
```

Japanese:
```
    "projects:manage:dialog:uninstall package as requested": "<b>{{name}}</b> をアンインストール",
    "projects:manage:dialog:uninstall package as legacy": "レガシーパッケージとして <b>{{name}}</b> をアンインストール",
    "projects:manage:dialog:uninstall package as unused": "使用されていない <b>{{name}}</b> をアンインストール",
```


I doubt that you ought to say "as you requested" or "since......". Based on Japanese localisation, we can simply say like below. What do you say to this?

```
    "projects:manage:dialog:uninstall package as requested": "Remove <b>{{name}}</b>.",
    "projects:manage:dialog:uninstall package as legacy": "Remove <b>{{name}}</b> as a legacy package.",
    "projects:manage:dialog:uninstall package as unused": "Remove <b>{{name}}</b> which is unused.",
```

**Reply by anatawa12** [reply]
Almost all parts of ALCOM is developed for English and then localized to Japanese by Sayamame so difference in "As requested" is problem or decision of Japanese localization.

I think difference between "since It's not used" and "使用されてない~" is not major problem, just a minor expression difference so it's not a problem.

**Reply by lu-na-r** [reply]
Absolutely, the minor difference is not a problem, but the expression of my proposal is more recognisable, isn't it? That's why I ask you. What do you think about that?

**Reply by anatawa12** [reply]
I think removing reason should be expressed so I don't like removing "as you requested” entirely.
I feel "as a legacy package" and "レガシーパッケージとして" is a little incorrect so I don't like to change it to "as a legacy package".
I think "which is unused" is good.

**Reply by lu-na-r** [reply]
Got it, thank you! I'll make a pull request about this later.

**Comment by Sayamame-beans** [comment]
`projects:tooltip:no directory` is the string that is displayed if the project was not found in the directory specified in the project information.
Currently, it indicates that the project directory does not exist, but this can be a bit confusing if the project path has simply changed. (e.g. parent directory is renamed)

How do you think about replacing it with something like, "The project was not found at this location"(this sample text is not well-considered) instead?

**Reply by lu-na-r** [reply]
Here're some ideas...

- `ALCOM cannot find this project at the registered directory.`
- `There is no project in the directory.`
- `The directory which you provided before is no longer a valid path.`

**Reply by anatawa12** [reply]
`Project does not exist at the location.`

**Reply by Sayamame-beans** [reply]
I feel `Project does not exist at the location.` represents the situation in a nutshell.

**Comment by anatawa12** [comment]
## The term `Resolve` in VPM system

In #930, I implemented detecting missing packages and ALCOM recommends the user to Resolve packages.
I want to discuss about the term "Resolve"

The "Resolve" means install all packages defined in `vpm-manifest.json` and it has several use cases.

- To install missing locked packages - When we cloned some package from VCS such as git, vpm packages (except for resolver) are missing in the project. Resolving packages will installs missing packages
- To install missing dependencies - The template only includes dependencies section. When we created project, the ALCOM will copy the template and resolve the new project to install packages.
- To fix broken package - When we modified files in the package, the package become broken. we can fix with resolving packages.

I want to use similar notation for all languages including english.
Therefore, I want to discuss about how translate it.

I have several idea:
- Use direct word translation for "Resolve". I already see "パッケージを解決する" for package-manager-generic verb so it might be possible however, it's not easy for users to understand.
- Say "there are problems" and "fix problem" button. This is easy to implement but it does not describe what to do.
- Say "recommend to install packages" and "install package" button. This is also easy to implement and it describe what to do.

Is there any ideas?

For information, in ALCOM 0.1.0, resolving packages was implemented as "reinstall packages" button for the third use case.

**Reply by JustBuddy** [reply]
I went through multiple iterations of this for the german translation.
The translations should state what the issue is, and that this issue should be resolved.
German, and I assume many other languages, depend on context. So the wording must fit the way sentences are built in that language.
My translation translates back to englisch as: The project requires packages which currently are not installed. This should be corrected. The german translation uses correction instead of solving, as building a sentence and providing the button for a more literal translation (Lösen) is a bit awkward.

If you reuse that string for those three use cases, the message should be either a bit more generalized or include all the possible cases, like: This project uses packages that are not installed or defective. [...]

As long as the message clearly conveys what it is for, it should be fine.

**Reply by Sayamame-beans** [reply]
I think it is better to avoid using "Resolve" to describe a situation.
On the other hand, "Resolve" for the button could be read as resolving the problem, as well as package resolving, so it may be well-balanced. (maybe?)

I feel it is a bit difficult to decide how to translate it in Japanese. ("解決" might be a bit stiff)

**Reply by lonelyicer** [reply]
Say "there are problems" and "fix problem" button. Then list the problems that have arisen

**Reply by anatawa12** [reply]
> If you reuse that string for those three use cases

I'll use the text added in #930 only for the first or possibly second use case of Resolve. so "The project requires packages which currently are not installed. This should be corrected." looks fine.

> Say "there are problems" and "fix problem" button. Then list the problems that have arisen

I'm not for adding dialog to show problems.
I saied 'Say "there are problems"' but I mean describe the problem. Sorry for confusing.

**Reply by anatawa12** [reply]
After thinking a while, I feel this is good. How do you think that?

<img width="600" alt="image" src="https://github.com/vrc-get/vrc-get/assets/22656849/332ae184-9b32-4ab9-b888-8764815603b5">

**Reply by lu-na-r** [reply]
"this project" rather than "the project"?

**Reply by anatawa12** [reply]
> "this project" rather than "the project"?

That's better. thank you

**Comment by JustBuddy** [comment]
@anatawa12 @Sayamame-beans Not a fan of `projects:type:custom` being "Custom", it is not immediately clear that this is for templates and confused me while translating this dialog. I propose something among the lines of "Custom Template" to make this association clear.

What do you think?

**Reply by anatawa12** [reply]
Using "Custom Template" looks good!

**Reply by lu-na-r** [reply]
Done in #973

**Reply by Sayamame-beans** [reply]
Reconsidering this, it would be better to use “projects:template:type:custom” or “projects:template:custom” since it is _still_ confusing with the other “projects:type:*”.
However, when #602 will be implemented, it may also be a good idea to reclassify it as a key of the area of 602.

**Comment by Sayamame-beans** [comment]
- #1002 
I'd like to add unity version information in `projects:toast:match version unity not found` as follows
```diff
- "No matching Unity version found. Please install Unity in your computer or add a Unity version in the ALCOM settings."
+ "Unity {{unity}} is not found. Please install Unity in your computer or add a Unity version in the ALCOM settings."
```

**Reply by Sayamame-beans** [reply]
This PR has been merged.

**Comment by JustBuddy** [comment]
I'd love to see the "Search for Updates" key split up a bit, this menu item looks a bit awkward right now, since caption and button text is the same.
"settings:check update"
![image](https://github.com/vrc-get/vrc-get/assets/8122264/c0a46f8f-d6cd-44bd-a7a9-3c26dcb1d83a)


I would like to see it more like:
"settings:check update" : "ALCOM Updates",
"settings:check update:button" : "Check for updates",

![image](https://github.com/vrc-get/vrc-get/assets/8122264/3f78fca6-1181-4c33-87d9-e419a974d937)

What do you think?

**Reply by Sayamame-beans** [reply]
Sounds good?
Also there's space to add an option to receive beta updates of ALCOM or not in the future... maybe?

**Reply by anatawa12** [reply]
I think it's good.

**Comment by Sayamame-beans** [comment]
Representations "non-ascii" and "multibyte char" should be unified?
e.g.
https://github.com/vrc-get/vrc-get/blob/dba937c72189b072948c23d64e2e16193219631d/vrc-get-gui/locales/en.json5#L256
https://github.com/vrc-get/vrc-get/blob/dba937c72189b072948c23d64e2e16193219631d/vrc-get-gui/locales/en.json5#L83

**Reply by Spokeek** [reply]
As it's mostly a warning and we don't take any specific action, i'd say yes, unite them.

**Reply by Sayamame-beans** [reply]
I reconsidered for this, project name validation has also warn for symbol (include ascii), so unify "multibyte" to non-ascii is not good.
So... I leave this as is.

**Comment by JustBuddy** [comment]
I'd propose a different way to display the new "open" buttons in the "ALCOM" section of the settings.

```
    "settings:button:open settings.json": "settings.json",
    "settings:button:open gui config.json": "gui-config.json",
    "settings:button:open logs": "Log Folder",
    "settings:button:open custom templates": "Project Templates", 
```

If that menu is designed in a way that makes it clear that clicking those opens things, we can reduce redundant "open" button text prefixes.
Here is how that looks in my translation:
![image](https://github.com/user-attachments/assets/23465e4c-98f8-4c5f-bab3-0e40c999ae19)

Reference VCC
![image](https://github.com/user-attachments/assets/ea89dccf-ef47-4995-8537-dbdaea522d09)

**Comment by Spokeek** [comment]
Regarding the translation guidelines, i see we recommend the use of os-independant language

> - ALCOM is cross-platform, so please use os-independent words as possible.
 >   - For example, use "Directory" instead of "Folder"

But i saw quite a few keys using the word Folder instead of Directory, should we change them all back to directory then ?

**Reply by anatawa12** [reply]
They are my mistake. we should fix soon.

**Comment by yelandiealma** [comment]
Hey, I think {{unity}} and {{chinaUnity}} might be reversed...?
https://github.com/vrc-get/vrc-get/blob/731ba58171cf113907e0e5a3014b0daa05ad7ade/vrc-get-gui/locales/en.json5#L143

**Reply by anatawa12** [reply]
Thank you for let us know.

It's not reversed in actual message, {{chinaUnity}} is used as {{internationalUnity}} but this is very confusing for translators.

https://github.com/vrc-get/vrc-get/blob/731ba58171cf113907e0e5a3014b0daa05ad7ade/vrc-get-gui/lib/use-open-unity.tsx#L346-L352


I think it's better to rename {{unity}} with {{expectedUnity}} (for both mesage) and {{chinaUnity}} with {{internationalUnity}} in this mesage so I will soon.
